storage/src/test/java/tech/pegasys/teku/storage/server/network/DatabaseNetworkTest.java

```diff
@@ -82,4 +86,45 @@ public void shouldNotThrowIfForkAndContractMatch(@TempDir final File tempDir) th
 
     assertDoesNotThrow(() -> DatabaseNetwork.init(networkFile, fork, eth1Address));
   }
+
+  @Test
+  void shouldWriteAndReadDatabaseNetworkWithChainId(@TempDir final File tempDir)
+      throws IOException {
+    final File networkFile = new File(tempDir, "network.yml");
+
+    final Bytes4 fork = dataStructureUtil.randomFork().getCurrentVersion();
+    final Eth1Address eth1Address = dataStructureUtil.randomEth1Address();
+    final Long depositId = dataStructureUtil.randomLong();
+    final DatabaseNetwork databaseNetwork =
+        new DatabaseNetwork(fork.toHexString(), eth1Address.toHexString(), depositId);
+
+    final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
```

Comment:
can be shared between tests, create with same option as the static function in real implementation